hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1769,7 +1769,7 @@ def accept(*args, **kwargs):
 def complex_numbers(
     *,
     min_magnitude: Optional[Real] = 0,
```

Comment:
```suggestion
    min_magnitude: Real = 0,
```

There's no real reason to accept `min_magnitude=None`, as there's a clear lower-bound on possible magnitudes.

We went through a similar transition with `min_size` arguments in #1618, but missed magnitudes... happy for you to just do the small fix above.  Alternatively you could update the function - inline the `check_valid_magnitude` logic, add a deprecation for `None`, and change `min_magnitude is None` to `min_magnitude == 0`.

Reply:
happy to change the `check_valid_magnitude` logic; do you prefer a separate PR or an addition into this (plus a rename)?

Reply:
Let's keep it in this one for simplicity - that way we're changing the type annotations to clarify the expected behaviour and the release notes can just say something like 

> This release deprecates passing `min_magnitude=None` to `complex_numbers()`.  If there should be no lower bound, you can pass `min_magnitude=0` or simply omit this argument.

Plus a similar message passed to `note_deprecation(msg_here, since="RELEASEDAY")` - our automation will replace that with the actual date on release :smile: